Fix test examples on windows

AlexanderDokuchaev opened the pull request:
### Changes

- `setuptools<=72.1.0` to avoid issue `Could not compile CPU quantization extensions. Falling back on torch native operations - CPU quantization fine-tuning may be slower than expected.` on windows.
- Add catch `BackendCompilerFailed` exception for `post_training_quantization_torch_fx_resnet18`
-  Bump version of anomalib to 1.0.1 `quantization_aware_training_torch_resnet18` tests on windows

### Related tickets

155866

AlexanderDokuchaev commented:
windows/job/test-examples/216/

alexsu52 reviewed (commented):
- quantization_aware_training_torch_resnet18 is regression.
- post_training_quantization_torch_fx_resnet18 is a new example and should be reworked to run on windows.

@AlexanderDokuchaev, please, file tickets to handle issues that were skipped.

cc' @MaximProshin

AlexanderDokuchaev commented:
> * quantization_aware_training_torch_resnet18 is regression.

It's not regression, 1.0.0 falls on training. 
Updated to 1.0.1 - windows/job/test-examples/219/ 
And anomalib required enabled develop mode on host or run test as administrator 

> * post_training_quantization_torch_fx_resnet18 is a new example and should be reworked to run on windows.

Added catch exception for BackendCompilerFailed
windows/job/test-examples/220

alexsu52 reviewed (approved):
LGTM
